src/tbp/monty/frameworks/models/evidence_matching/evidence_updater.py

```diff
@@ -0,0 +1,782 @@
+            past_weight (float): How much should the evidence accumulated so far be
+                weighted when combined with the evidence from the most recent
+                observation. Defaults to 1.
+            present_weight (float): How much should the current evidence be weighted
+                when added to the previous evidence. If past_weight and present_weight
+                add up to 1, the evidence is bounded and can't grow infinitely. Defaults
+                to 1.
+                NOTE: right now this doesn't give as good performance as with unbounded
+                evidence since we don't keep a full history of what we saw. With a more
+                efficient policy and better parameters that may be possible to use
+                though and could help when moving from one object to another and to
+                generally make setting thresholds etc. more intuitive.
+            x_percent_threshold (float): Used in two places:
+                1) All objects whose highest evidence is greater than the most likely
+                    objects evidence - x_percent of the most like objects evidence are
+                    considered possible matches. That means to only have one possible
+                    match, no other object can have more evidence than the candidate
+                    match's evidence - x percent of it.
+                2) Within one object, possible poses are considered possible if their
+                    evidence is larger than the most likely pose of this object - x
+                    percent of this poses evidence.
+                TODO: should we use a separate threshold for within and between objects?
+                If this value is larger, the model is usually more robust to noise and
+                reaches a better performance but also requires a lot more steps to reach
+                a terminal condition, especially if there are many similar object in the
+                data set. Defaults to 10.
+        """
+        self.evidence_update_threshold = evidence_update_threshold
+        self.feature_evidence_increment = feature_evidence_increment
+        self.feature_weights = feature_weights
+        self.graph_memory = graph_memory
+        self.initial_possible_poses = get_initial_possible_poses(initial_possible_poses)
+        self.max_match_distance = max_match_distance
+        self.max_nneighbors = max_nneighbors
+        self.past_weight = past_weight
+        self.present_weight = present_weight
+        self.tolerances = tolerances
+        self.x_percent_threshold = x_percent_threshold
+
+        self.use_features_for_matching = self._check_use_features_for_matching()
+
+    def update_evidence(
+        self,
+        evidence: np.ndarray,
+        features: dict,
+        displacements: dict | None,
+        graph_id: str,
+        locations: np.ndarray,
+        mapper: ChannelMapper,
+        poses: np.ndarray,
```

Comment:
again here, the variable names are kind of confusing. Which of those refer to hypotheses and which refer to the current sensation? They should at least be grouped together but idealy this information should also be in the variable name